SMTChecker: Bring back assertions in "external_calls" SMTChecker tests (#15204)
ethereum/solidity: docs/installing-solidity.rst

SMT Solvers
-----------
Solidity can be built against Z3 SMT solver and will do so by default if
it is found in the system. Z3 can be disabled by a ``cmake`` option.

*Note: In some cases, this can also be a potential workaround for build failures.*

Inside the build folder you can disable Z3, since it is enabled by default:

.. code-block:: bash

    # disables Z3 SMT Solver.
    cmake .. -DUSE_Z3=OFF

*Note*: Solidity can optionally use other solvers, namely ``cvc5`` and ``Eldarica``,

> but their presence is checked during runtime, the build is not affected.

Author: Suggestions for wording welcome!

> *Note*: Solidity can optionally use other solvers, namely ``cvc5`` and ``Eldarica``,
> but their presence is checked during runtime, the build is not affected.

Reviewer: ```suggestion
.. note::

    Solidity can optionally use other solvers, namely ``cvc5`` and ``Eldarica``,
    but their presence is checked during runtime, the build is not affected.
```

SMT Solvers
-----------
Solidity can be built against Z3 SMT solver and will do so by default if
it is found in the system. Z3 can be disabled by a ``cmake`` option.

*Note: In some cases, this can also be a potential workaround for build failures.*

Inside the build folder you can disable Z3, since it is enabled by default:

.. code-block:: bash

    # disables Z3 SMT Solver.
    cmake .. -DUSE_Z3=OFF

.. note::

>     Solidity can optionally use other solvers, namely ``cvc5`` and ``Eldarica``,
>     but their presence is checked during runtime, the build is not affected.

Reviewer 2: ```suggestion
    Solidity can optionally use other solvers, namely ``cvc5`` and ``Eldarica``,
    but their presence is checked only at runtime, they are not needed for the build to succeed.
```
